src/buffer/out/textBuffer.cpp

```diff
@@ -1186,6 +1189,51 @@ const COORD TextBuffer::_GetWordEndForSelection(const COORD target, const std::w
     return result;
 }
 
+void TextBuffer::_PruneHyperlinks()
+{
+    // First, check the old first row for hyperlink references
+    // If there are any, search the entire buffer for the same reference
+    // If the buffer does not contain the same reference, we can remove that hyperlink from our map
+    // This way, obsolete hyperlink references are cleared from our hyperlink map instead of hanging around
+    std::unordered_set<USHORT> refs;
+    const auto firstAttrRow = _storage.at(_firstRow).GetAttrRow();
+    // Get all the hyperlink references in the row we're erasing
+    for (auto it = firstAttrRow.begin(); it != firstAttrRow.end(); ++it)
+    {
+        if (it->IsHyperlink())
+        {
+            refs.emplace(it->GetHyperlinkId());
+        }
+    }
+    if (!refs.empty())
+    {
+        const auto total = TotalRowCount();
+        // Loop through all the rows in the buffer
+        for (size_t i = 1; i != total; ++i)

```

Comment:
I know `GetRowByOffset` is a mod operation so you're touching every row, but why is this from 1 to Total instead of 0 to (Total-1)? Leave a comment?

Reply:
I don't wanna to check the first row since that is the row that is going to be erased and moved to the bottom of the buffer

Reply:
I believe Michael's suggestion is "leave a comment in the code explaining it" :smile:

Reply:
Apologies for being obtuse.